vdyp-sindex/sindxdll.java

```diff
@@ -0,0 +1,3947 @@
+    case SI_SPEC_OG: return 0x00;
+    case SI_SPEC_P: return 0x02;
+    case SI_SPEC_PA: return 0x02;
+    case SI_SPEC_PF: return 0x02;
+    case SI_SPEC_PJ: return 0x02;
+    case SI_SPEC_PL: return 0x06;
+    case SI_SPEC_PLC: return 0x01;
+    case SI_SPEC_PLI: return 0x06;
+    case SI_SPEC_PM: return 0x00;
+    case SI_SPEC_PR: return 0x00;
+    case SI_SPEC_PS: return 0x00;
+    case SI_SPEC_PW: return 0x04;
+    case SI_SPEC_PXJ: return 0x02;
+    case SI_SPEC_PY: return 0x06;
+    case SI_SPEC_Q: return 0x00;
+    case SI_SPEC_QE: return 0x00;
+    case SI_SPEC_QG: return 0x01;
+    case SI_SPEC_R: return 0x01;
+    case SI_SPEC_RA: return 0x01;
+    case SI_SPEC_S: return 0x00;
+    case SI_SPEC_SA: return 0x02;
+    case SI_SPEC_SB: return 0x06;
+    case SI_SPEC_SE: return 0x06;
+    case SI_SPEC_SI: return 0x02;
+    case SI_SPEC_SN: return 0x02;
+    case SI_SPEC_SS: return 0x05;
+    case SI_SPEC_SW: return 0x06;
+    case SI_SPEC_SX: return 0x06;
+    case SI_SPEC_SXB: return 0x02;
+    case SI_SPEC_SXE: return 0x01;
+    case SI_SPEC_SXL: return 0x01;
+    case SI_SPEC_SXS: return 0x01;
+    case SI_SPEC_SXW: return 0x02;
+    case SI_SPEC_SXX: return 0x02;
+    case SI_SPEC_T: return 0x00;
+    case SI_SPEC_TW: return 0x00;
+    case SI_SPEC_U: return 0x00;
+    case SI_SPEC_UA: return 0x00;
+    case SI_SPEC_UP: return 0x00;
+    case SI_SPEC_V: return 0x00;
+    case SI_SPEC_VB: return 0x00;
+    case SI_SPEC_VP: return 0x00;
+    case SI_SPEC_VS: return 0x00;
+    case SI_SPEC_VV: return 0x00;
+    case SI_SPEC_W: return 0x00;
+    case SI_SPEC_WA: return 0x00;
+    case SI_SPEC_WB: return 0x00;
+    case SI_SPEC_WD: return 0x00;
+    case SI_SPEC_WI: return 0x00;
+    case SI_SPEC_WP: return 0x00;
+    case SI_SPEC_WS: return 0x00;
+    case SI_SPEC_WT: return 0x00;
+    case SI_SPEC_X: return 0x00;
+    case SI_SPEC_XC: return 0x00;
+    case SI_SPEC_XH: return 0x00;
+    case SI_SPEC_Y: return 0x00;
+    case SI_SPEC_YC: return 0x01;
+    case SI_SPEC_YP: return 0x00;
+    case SI_SPEC_Z: return 0x00;
+    case SI_SPEC_ZC: return 0x00;
+    case SI_SPEC_ZH: return 0x00;
+    }
+
+return SI_ERR_SPEC;
+}
+
+
+/*-----------------------------------------------------------------------------
+ *
+ *  Sindex_DefCurve
+ *  ===============
+ *    Returns default curve index for a species.
+ *
+ *  Parameters
+ *  ----------
+ *    Integer species index.
+ *
+ *  Return Value
+ *  ------------
+ *    Integer curve index, for use in other Sindex functions.
+ *    May return an error code under the following conditions:
+ *    
+ *    return value    condition
+ *    ------------    ---------
+ *    SI_ERR_SPEC     input parameter is not a valid species index
+ *    SI_ERR_NO_ANS   no curves defined for this species
+ ----------------------------------------------------------------------------- */
+public static short Sindex_DefCurve(short sp_index) {
+    if (sp_index < 0 || sp_index >= SI_MAX_SPECIES){
+      return SI_ERR_SPEC;

```

Comment:
Switching to Java exceptions instead of returning error codes might be a good idea in the long run.